src/core/vector/qgsvectorlayer.cpp

```diff
@@ -297,6 +297,11 @@ QgsVectorLayer *QgsVectorLayer::clone() const
     layer->dataProvider()->handlePostCloneOperations( mDataProvider );
   }
   QgsMapLayer::clone( layer );
+  layer->mXmlExtent2D = mXmlExtent2D;
+  layer->mLazyExtent2D = mLazyExtent2D;
+  layer->mValidExtent2D = mValidExtent2D;
+  layer->mLazyExtent3D = mLazyExtent3D;
```

Comment:
probably needs to clone mXmlExtent3D too ?
```suggestion
  layer->mXmlExtent3D = mXmlExtent3D;
  layer->mLazyExtent3D = mLazyExtent3D;
```